libs/server/Resp/BasicCommands.cs

```diff
@@ -366,24 +332,18 @@ private bool NetworkSETEX<TGarnetApi>(bool highPrecision, ref TGarnetApi storage
             }
 
             var val = parseState.GetArgSliceByRef(2).SpanByte;
-            var valPtr = val.ToPointer() - (sizeof(int) + sizeof(long));
-            var vSize = val.Length;
 
-            // Save prior state on network buffer
-            var save1 = *(int*)valPtr;
-            var save2 = *(long*)(valPtr + sizeof(int));
+            var newValLen = val.Length + sizeof(long);
+            var newVal = stackalloc byte[newValLen];
 
-            *(int*)valPtr = vSize + sizeof(long); // expiry info
-            SpanByte.Reinterpret(valPtr).ExtraMetadata = DateTimeOffset.UtcNow.Ticks +
-                                                         (highPrecision
-                                                             ? TimeSpan.FromMilliseconds(expiry).Ticks
-                                                             : TimeSpan.FromSeconds(expiry).Ticks);
+            var sbNewVal = SpanByte.FromPinnedPointer(newVal, newValLen);
+            sbNewVal.ExtraMetadata = DateTimeOffset.UtcNow.Ticks +
+                                     (highPrecision
+                                         ? TimeSpan.FromMilliseconds(expiry).Ticks
+                                         : TimeSpan.FromSeconds(expiry).Ticks);
+            val.AsReadOnlySpan().CopyTo(sbNewVal.AsSpan());
```

Comment:
We cannot afford this copy. The item is already in the network buffer, we just need to add the expiration header. Perhaps we could add an `SessionParseState.UnsafeInsertMetadata(int offset, long metadata)` which adjusts the specified ArgSlice with the assumption that the caller has ensured there is enough space for it in the buffer. When we move to `RawStringInput` for Upsert, this will no longer be necessary.

Reply:
A safer alternative is to send only the expiration via RawStringInput (arg1) in the call to Upsert, even now.